### PR TITLE
[glibmm] modernize

### PR DESCRIPTION
**Pull request by planetmarshall**
Specify library name and version:  **glibmm/**

Modernize recipe and bump versions

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by planetmarshall** [comment]
Missing a MTd/Shared build of glib

**Comment by uilianries** [comment]
@planetmarshall MT + Shared is no longer built and will not: https://github.com/conan-io/conan-center-index/blob/master/docs/supported_platforms_and_configurations.md#windows

**Comment by planetmarshall** [comment]
> @planetmarshall MT + Shared is no longer built and will not: https://github.com/conan-io/conan-center-index/blob/master/docs/supported_platforms_and_configurations.md#windows

OK - so why is it looking for that configuration?

```
ERROR: Missing binary: glib/2.73.3:4018e68a0242b05a4bed65d90f5f3255a7880b6c
glib/2.73.3: WARN: Can't find a 'glib/2.73.3' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Debug, compiler=Visual Studio, compiler.runtime=MTd, compiler.version=15, os=Windows
- Options: shared=True...
```

**Comment by uilianries** [comment]
@planetmarshall No idea, need to be investigated ...

**Comment by planetmarshall** [comment]
> @planetmarshall No idea, need to be investigated ...

OK - this is a knock on effect of making `glib` shared by default. It's building a static configuration of `glibmm` with `MTd` so is looking for shared `glib` with the same runtime which doesn't exist.

I think we will have to make static glibmm with shared glib and `MTd` an unsupported configuration.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/glibmm/all/conanfile.py [hunk 1]
```diff
@@ -138,104 +148,54 @@ def package(self):
...
+        self.cpp_info.components[glibmm_component].libs = [glibmm_component]
+        self.cpp_info.components[glibmm_component].includedirs = [os.path.join("include", glibmm_component)]
 
+        if self._abi_version == "2.68":
+            self.cpp_info.components[glibmm_component].requires = ["glib::gobject-2.0", "libsigcpp::sigc++"]
         else:
-            self.cpp_info.components["glibmm-2.4"].names[
-                "pkg_config"] = "glibmm-2.4"
-            self.cpp_info.components["glibmm-2.4"].libs = ["glibmm-2.4"]
-            self.cpp_info.components["glibmm-2.4"].libs = ["glibmm-2.4"]
```
Comment (prince-chrismc) [inline recipes/glibmm/all/conanfile.py:222]: Am I crazy or were these duplicates?

Comment (planetmarshall) [reply recipes/glibmm/all/conanfile.py:222]: Not crazy :)

**Pull request merged**